packages/sdk-js/src/GrowthBook.ts

```diff
@@ -1149,4 +1297,151 @@ export class GrowthBook<
+            if (rule.fallbackAttribute) {
+              attributes.add(rule.fallbackAttribute);
+            }
+          }
+        }
+      }
+    });
+    experiments.map((experiment) => {
+      attributes.add(experiment.hashAttribute || "id");
+      if (experiment.fallbackAttribute) {
+        attributes.add(experiment.fallbackAttribute);
+      }
+    });
+    return Array.from(attributes);
+  }
+
+  public async refreshStickyBuckets(data?: FeatureApiResponse) {
+    if (this.context.stickyBucketService) {
+      const attributes = this._getStickyBucketAttributes(data);
+      this.context.stickyBucketAssignmentDocs = await this.context.stickyBucketService.getAllAssignments(
+        attributes
+      );
+    }
+  }
+
+  private _getStickyBucketAssignments(): StickyAssignments {
+    const mergedAssignments: StickyAssignments = {};
+    Object.entries(this.context.stickyBucketAssignmentDocs ?? {}).forEach(
+      ([, doc]) => {
+        if (doc.assignments) Object.assign(mergedAssignments, doc.assignments);
+      }
+    );
+    return mergedAssignments;
+  }
+
+  private _getStickyBucketVariation(
+    experimentKey: string,
+    experimentBucketVersion: number = 1,
+    minExperimentBucketVersion: number = 0,
+    meta: VariationMeta[] = []
+  ): {
+    variation: number;
+    versionIsBlocked: boolean;
+  } {
+    const id = this._getStickyBucketExperimentKey(
+      experimentKey,
+      experimentBucketVersion
+    );
+    const assignments = this._getStickyBucketAssignments();
+
+    // users with any blocked bucket version (0 to minExperimentBucketVersion) are excluded from the test
+    if (minExperimentBucketVersion > 0) {
+      for (let i = 0; i <= minExperimentBucketVersion; i++) {
+        const blockedKey = this._getStickyBucketExperimentKey(experimentKey, i);
+        if (assignments[blockedKey] !== undefined) {
+          return {
+            variation: -1,
+            versionIsBlocked: true,
+          };
+        }
+      }
+    }
+    const variationKey = assignments[id];
+    if (variationKey === undefined)
+      // no assignment found
+      return {
+        variation: -1,
+        versionIsBlocked: false,
+      };
+    const variation = meta.findIndex((m) => m.key === variationKey);
+    if (variation < 0)
+      // invalid assignment, treat as "no assignment found"
+      return {
+        variation: -1,
+        versionIsBlocked: false,
+      };
+
+    return { variation, versionIsBlocked: false };
+  }
+
+  private _getStickyBucketExperimentKey(
+    experimentKey: string,
+    experimentBucketVersion: number = 1
+  ): StickyExperimentKey {
+    return `${experimentKey}__${experimentBucketVersion}`;
+  }
+
+  private _getStickyBucketAttributes(
+    data?: FeatureApiResponse
+  ): Record<string, string> {
+    const attributes: Record<string, string> = {};
+    const shouldDeriveIdentifierAttributes =
+      data || this.context.stickyBucketIdentifierAttributes === undefined;
```

Comment:
Does the `data ||` part need to be here?  That implies it will overwrite the explicitly set context options, but it feels like explicitly set options should always take precedence over derived attributes.